mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixture.scala

```diff
@@ -121,6 +124,17 @@ class GaussianMixture private (
    */
   def getConvergenceTol: Double = convergenceTol
 
+  /**
+   * Sets indicator for computing Gaussians on driver or executors
+   */
+  def setDistributeGaussians(distributeGaussians: Boolean): this.type = {
```

Comment:
OK, using the heuristic dimension >= 30 and numClusters >= 10 based on initial [perf tests](https://gist.github.com/feynmanliang/70d79c23dffc828939ec)